src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java

```diff
@@ -77,6 +85,17 @@
     @Argument(fullName = "gcs_max_retries", shortName = "gcs_retries", doc = "If the GCS bucket channel errors out, how many times it will attempt to re-initiate the connection", optional = true)
     public int NIO_MAX_REOPENS = BucketUtils.DEFAULT_GCS_MAX_REOPENS;
 
+    // This option is here for documentation completeness.
+    // This is actually parsed out in Main to initialize configuration files because
+    // we need to have the configuration completely set up before we create our CommandLinePrograms.
+    // (Some of the CommandLinePrograms have default values set to config values, and these are loaded
+    // at class load time as static initializers).
+    @Argument(fullName = StandardArgumentDefinitions.GATK_CONFIG_FILE_OPTION,
```

Comment:
Now downstream projects extending `CommandLineProgram` would have this option, but because it is not really used in the class, the final user may have an unused argument that it is documented (in the CLI and Barclay) but not having any effect.

I want to do not allow the final user to provide a config file by now, but the users of my software may ask me for it if they see this option. Thus, I have two suggestions for this argument:

* `CommandLineProgramConfigOptions` interface for an argument container for static configuration of the framework. This may contain arguments purely for documentation (such as the config file), but also can include a method (`setup()`) for setting up static stuff (e.g., `BlockCompressedOutputStream` inflater/deflater factory, temp directory, etc.). A `CommandLineProgram.getConfigArgs()`  will return a default `CommandLineProgramConfigOptionsGATK` containing the config file option for documentation purposes, and the setup arguments for temp directories, verbosity, deflater/inflater, GCS retries, etc. Users may override this method in an simple extension class for the walkers that they use, to provide a custom one to hide some variables or setup their environments in a different way.
* A different approach is similar to the previous one, but with a container only for documentation purposes options such the config file. Users may provide their own only-documentation options that won't be used but show in the help...

Reply:
As I said in the previous comment, this is bad for downstream project that does not want to expose a config file to users. Can be done in a different way?

E.g., overridable argument collection for config params - may include others (such as gcs_max_retries, etc...)